src/qtt/measurements/scans.py

```diff
@@ -1264,8 +1264,12 @@ def get_sampling_frequency(instrument_handle):
     if isinstance(instrument_handle, AcquisitionScopeInterface):
         return instrument_handle.sample_rate
     try:
+        import qcodes_contrib_drivers.drivers.Spectrum.M4i
+        if isinstance(qcodes_contrib_drivers.drivers.Spectrum.M4i):
+            return instrument_handle.sample_rate()
         import qcodes.instrument_drivers.Spectrum.M4i
         if isinstance(instrument_handle, qcodes.instrument_drivers.Spectrum.M4i.M4i):
+            warnings.warn('please use the M4i driver from qcodes_contrib_drivers')
```

Comment:
To make it possible to use the old m4i driver (when package contrib drivers is not installed) but with printing the warning you could do:

    try:
        import qcodes_contrib_drivers.drivers.Spectrum.M4i
        if isinstance(qcodes_contrib_drivers.drivers.Spectrum.M4i):
            return instrument_handle.sample_rate()
    except ImportError:
        pass
    try:
        import qcodes.instrument_drivers.Spectrum.M4i
        if isinstance(instrument_handle, qcodes.instrument_drivers.Spectrum.M4i.M4i):
            warnings.warn('please use the M4i driver from qcodes_contrib_drivers')
            return instrument_handle.sample_rate()
    except ImportError:
        pass